File: backend/src/main/java/com/tamakara/litebooth/domain/entity/Order.java
package com.tamakara.litebooth.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.Instant;

@Data
@TableName("`order`")
public class Order {
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    @TableField(value = "status")
    private String status;

    @TableField(value = "query_email")
    private String queryEmail;

    @TableField(value = "query_password")
    private String queryPassword;

    @TableField(value = "item_id")
    private Long itemId;

    @TableField(value = "item_name")
    private String itemName;

    @TableField(value = "item_price")
    private Double itemPrice;

    @TableField(value = "quantity")
    private Long quantity;

    @TableField(value = "amount")
    private Double amount;

    @TableField(value = "payment_method")
    private String paymentMethod;

    @TableField(value = "created_at")
    private Instant createdAt;

    @TableField(value = "payment_at")
    private Instant paymentAt;
}

File: frontend-admin/src/api/order.ts
import { http } from "@/utils/http";

export interface OrderVO {
  id: number;
  status: string;
  queryEmail: string;
  itemId: number;
  itemName: string;
  itemPrice: number;
  quantity: number;
  amount: number;
  paymentMethod: string;
  createdAt: string;
  paymentAt: string;
}

export interface OrderPageVO {
  records: OrderVO[];
  total: number;
  size: number;
  current: number;
}

export interface OrderPageQueryDTO {
  pageNum: number;
  pageSize: number;
  keyword?: string;
  status?: string;
}

export const getOrderPageVO = (data: OrderPageQueryDTO) => {
  return http.request<OrderPageVO>("post", "/order/getOrderPageVO", { data });
};

